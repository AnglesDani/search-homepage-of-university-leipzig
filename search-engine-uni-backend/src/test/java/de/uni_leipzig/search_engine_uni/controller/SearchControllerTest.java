package de.uni_leipzig.search_engine_uni.controller;

import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;

public class SearchControllerTest
{
	@Before
	public void setup()
	{
		SearchControllerMockUtils.setupHttpServlet();
	}
	
	@Test
	public void approveNullInput()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(0).search(null, null));
	}
	
	@Test
	public void approveEmptyQueryInput()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(0).search("", null));
	}
	
	@Test
	public void approveFirstPageResultWithOneMorePageAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 1));
	}
	
	@Test
	public void approveSecondPageResultWithOnePageBeforeAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 2));
	}
	
	@Test
	public void approveRequestOfFifthPageReturnsSecondPageIfOnlyElevenResultsAreAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 5));
	}
	
	@Test
	public void approveNextAndPreviousPageLinksAreCorrectForSecondPageOfThreePages()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(30).search("queryDoesNotMatter", 2));
	}
	
	@Test
	public void approveNegativePageRequestForFiftySevenAvailableResults()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(57).search("queryDoesNotMatter", -1));
	}
	
	@Test
	public void approveLastPageRequestForFiftySevenAvailableResults()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(57).search("queryDoesNotMatter", 6));
	}
}
