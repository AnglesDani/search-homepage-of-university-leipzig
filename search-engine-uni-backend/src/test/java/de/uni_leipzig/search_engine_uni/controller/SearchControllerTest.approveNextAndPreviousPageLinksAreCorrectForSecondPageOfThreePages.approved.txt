SearchResultPage(originalQuery=queryDoesNotMatter, page=2, resultsOnPage=[SearchResult(title=10, snippet=null, targetUrl=<http://www.example.com/url?documentId=10>;rel="targetUrl"), SearchResult(title=11, snippet=null, targetUrl=<http://www.example.com/url?documentId=11>;rel="targetUrl"), SearchResult(title=12, snippet=null, targetUrl=<http://www.example.com/url?documentId=12>;rel="targetUrl"), SearchResult(title=13, snippet=null, targetUrl=<http://www.example.com/url?documentId=13>;rel="targetUrl"), SearchResult(title=14, snippet=null, targetUrl=<http://www.example.com/url?documentId=14>;rel="targetUrl"), SearchResult(title=15, snippet=null, targetUrl=<http://www.example.com/url?documentId=15>;rel="targetUrl"), SearchResult(title=16, snippet=null, targetUrl=<http://www.example.com/url?documentId=16>;rel="targetUrl"), SearchResult(title=17, snippet=null, targetUrl=<http://www.example.com/url?documentId=17>;rel="targetUrl"), SearchResult(title=18, snippet=null, targetUrl=<http://www.example.com/url?documentId=18>;rel="targetUrl"), SearchResult(title=19, snippet=null, targetUrl=<http://www.example.com/url?documentId=19>;rel="targetUrl")], totalHits=30, nextPage=<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=3>;rel="next", previousPage=<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=1>;rel="prev", firstPageLink=null, lastPageLink=null, namedPageLinksBefore=[<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=1>;rel="1"], namedPageLinksAfter=[<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=3>;rel="3"])